test/unit/utils/params.cpp

```diff
@@ -0,0 +1,212 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <cstddef>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "ur_api.h"
+#include "ur_params.hpp"
+
+template <typename T> struct WrappedParams {
+    typedef T value_type;
+    virtual ~WrappedParams() {}
+    virtual T *get_struct() = 0;
+    virtual const char *get_expected() = 0;
+};
+
+template <typename T>
+std::unique_ptr<WrappedParams<typename T::value_type>> createParams();
+
+template <typename T> class ParamsTest : public testing::Test {
+  protected:
+    ParamsTest() : params(createParams<T>()) {}
+    ~ParamsTest() override {}
+
+    std::unique_ptr<WrappedParams<typename T::value_type>> params;
```

Comment:
It might make sense to have a non-template base class so that you don't have to use TYPED_TEST_SUITE but I don't see any benefit in using a template base class (WrappedParams).

Why not just store `T` directly here and in tests, just call `params.get_struct()`/`params.get_expected()`?

Reply:
I'm not sure what you mean exactly. I could store just T here (I'd also need the expected string, but that's a minor detail), but that wouldn't help with getting rid of TYPED_TEST_SUITE.
Theoretically, I could change it so that I just store strings here, and that would let me remove TYPED_TEST_SUITE, but I'm not sure how that is better.

Reply:
I simplified the tests.